### PR TITLE
Fix osgi.ee=UNKNOWN capability requirement

### PR DESCRIPTION
**Pull request by berezovskyi**
## Description

p2-maven-plugin since 1.7.0 correctly detects the bytecode level using
latest JDKs again

## Checklist

- [x] This PR adds an entry to the CHANGELOG. _See https://keepachangelog.com/en/1.0.0/ for instructions. Minor edits are exempt. 
- [ ] This PR expects some manual changes to the generated code. The needed changes are detailed in the CHANGELOG entry.
- [x] This PR was tested with at least one code generation, resulting in a working OSLC server. **Having Acceleo problems, see below**
- [x] This PR does NOT break the user block code

## Issues

Closes #0; Closes #00

_(use exactly this syntax to link to issues, one issue per statement only!)_

### CODE REVIEW

**Comment by berezovskyi** [comment]
We still have a minor problem with versions but at least LyoD runs now:

Screenshot from 2022-04-25 21-56-33

We will need to investigate the first problem and the rest I am thinking it's due to the mismatch of the RCP and Orbit P2 repos (I did not pull your changes today to update RCP from 2020-09 to 2021-03 to match the Orbit version).

**Comment by jadelkhoury** [comment]
we are packaging [com.google.gson_2.9.0.jar](https://download.eclipse.org/lyo/bundle/p2/edge/plugins/com.google.gson_2.9.0.jar)

See https://download.eclipse.org/lyo/bundle/p2/edge/plugins/

**Comment by jadelkhoury** [comment]
I tested this branch with older versions of the plugins (see pic below), and it ran fine.
I then upgraded to the latest bundles, but then I got many "missing osgi-ee" (see 2nd pic below).

I am running on Eclipse 2020-12. Maybe time to run on a later eclipse as well?

image

image

**Comment by jadelkhoury** [comment]
In the last commit, I change the build to be bsaed on Eclipse 2021-09. This results in following instead


image

**Comment by jadelkhoury** [comment]
With the last commit:
1. I build the bundles using 2022-03
2. I run from Eclipse 2022-03
Now we are down to the problem with Gson version only.

image

**Comment by jadelkhoury** [comment]
@berezovskyi ! I cannot add you as a reviewer, but I have a working solution. Can you please review to see what changes I made?
Please note that this upgrades LyoD to use eclipse 2022-03! It also means we will upgrade to latest version of Sirius and Acceleo. (Maybe it was not necessary, but might as well)

The mail problem is that we need to build Lyod based on the bundles we construct within that same build (and not what is under https://download.eclipse.org/lyo/bundle/p2/edge/

File: pom.xml [hunk 1]
```diff
@@ -11,8 +11,8 @@
         <tycho.version>2.2.0</tycho.version>
         <eclipse.cbi>1.3.2</eclipse.cbi>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <p2.repo>http://download.eclipse.org/releases/2020-12</p2.repo>
-        <lyo.repo>https://download.eclipse.org/lyo/bundle/p2/edge/</lyo.repo>
+        <p2.repo>http://download.eclipse.org/releases/2022-03</p2.repo>
+        <lyo.repo>file:org.eclipse.lyo.oslc4j.plugins/target/repository</lyo.repo>
```
Comment (berezovskyi) [inline pom.xml:15]: This is brilliant!

**Review by berezovskyi — COMMENTED** [review]
Great, I will test it on 2022-03 today. Thank you!

**Comment by jadelkhoury** [comment]
I've already tested, but please do to double check if you want.

**Comment by jadelkhoury** [comment]
Fails to build https://ci.eclipse.org/lyo/job/lyo-designer-test/55/console

Need to test locally again with same maven version 3.8.2

**Comment by jadelkhoury** [comment]
And now I can build on Jenkins, by introducing a new mvn build step that simply builds the "plugins" module first (everything else is then rebuilt again)
See https://ci.eclipse.org/lyo/job/lyo-designer-test/configure

New exe from: https://download.eclipse.org/lyo/product/binaries/superedge/

Update existing installation with: https://download.eclipse.org/lyo/product/p2/superedge

**Comment by berezovskyi** [comment]
Looks good to me, let's merge!

Did I understand you right that you want to change `file:org.eclipse.lyo.oslc4j.plugins/target/repository` somehow? I think I actually like this approach better as it's more deterministic.

FYI, I still have the UNKNOWN warning about a JSON-LD library (though everything seems to work at runtime):

Screenshot from 2022-04-27 23-53-24

**Comment by berezovskyi** [comment]
Actually, in the CI build, Equinox (during a Tycho build) complains that "file:org.eclipse.lyo.oslc4j.plugins/target/repository: URI is not hierarchical".

**Comment by berezovskyi** [comment]
Just updated the Github Action to match the build steps in Eclipse Jenkins.

**Comment by jadelkhoury** [comment]
> FYI, I still have the UNKNOWN warning about a JSON-LD library (though everything seems to work at runtime):

From the exe files you downloaded? Or building your own? 
If building your own, are you using 2022-03?

**Comment by jadelkhoury** [comment]
> Actually, in the CI build, Equinox (during a Tycho build) complains that "file:org.eclipse.lyo.oslc4j.plugins/target/repository: URI is not hierarchical".

Can you share a link to see where? If this happens, it means it is not building correctly. This should mean the exe won't work. But it does!!!

**Comment by berezovskyi** [comment]
>  If building your own, are you using 2022-03?

Running from the source, Eclipse 2022-03.

> Can you share a link to see where? If this happens, it means it is not building correctly. This should mean the exe won't work. But it does!!!

That's not a problem on my machine but instead on the CI from Github Actions. It was showing PR as failing to build. I added a build step for the bundles, copying what you did in the Jenkins build and now it works well: [`f240063` (#252)](https://github.com/eclipse/lyo.designer/pull/252/commits/f240063a66e30a3ba2e0a515f2e5fb2a282f1904)

**Review by jadelkhoury — APPROVED** [review]

**Pull request merged**